### PR TITLE
Method shorthand rules

### PR DESCRIPTION
**Pull request by kswilster**
Lint rule enforcing method shorthand.

prime: @dfreeman

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,6 +10,7 @@ module.exports = {
       "object": true,
       "block": true
     }],
-    "space-before-function-params": [2, false]
+    "space-before-function-params": [2, false],
+    "function-shorthand": [2, true]
```
Comment (dfreeman) [inline index.js]: This name feels a little ambiguous to me -- is it talking about `() => true` style or `{ method() {} }` style?
The terminology I typically see for those features is 'arrows' for the former and 'method shorthand' for the latter.

Comment (kswilster) [inline index.js]: valid

File: rules/function-shorthand.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        context.report(funct, 'Functions {{should}} use ES6 shorthand', {
+          should: shouldUseShorthand ? 'should' : 'should not'
+        });
+      }
+    }
+  }
+
+  return {
+    FunctionDeclaration: check,
+    FunctionExpression: check
```
Comment (dfreeman) [inline rules/function-shorthand.js]: `Property` nodes actually have a `method` flag that might be easier to use for this rule. Then you can just check that flag, and whether the key is a `Literal` or `Identifier`, and report based on that.

```json
{
    "type": "Property",
    "key": {
        "type": "Literal",
        "value": "foo bar",
        "raw": "\"foo bar\""
    },
    "value": {
        "type": "FunctionExpression",
        "id": null,
        "params": [],
        "defaults": [],
        "body": {
            "type": "BlockStatement",
            "body": []
        },
        "rest": null,
        "generator": false,
        "expression": false
    },
    "kind": "init",
    "method": true,
    "shorthand": false,
    "computed": false
}
```

File: rules/method-shorthand.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  function check(node) {
+    var key = node.key.type;
+    var value = node.value.type;
+
+    if ((value === 'FunctionExpression')) {
+      // we only care about non-event methods
+      if (key === 'Identifier') {
+        if (!node.method) {
+          context.report(node, 'Functions should use ES6 shorthand');
```
Comment (dfreeman) [inline rules/method-shorthand.js]: Might say 'methods' instead for consistency

**Comment by dfreeman** [comment]
:ship:

**Comment by dfreeman** [comment]
Closed by 22894c145b178709ae6774adb65ceab6053e6d45

**Pull request closed without merging**